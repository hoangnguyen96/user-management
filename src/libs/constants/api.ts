export const API_NAME = "user_management";

export enum ENDPOINTS {
  USER = "user",
  PRODUCT = "product",
}

export enum METHOD {
  GET = "GET",
  POST = "POST",
  PUT = "PUT",
  PATCH = "PATCH",
  DELETE = "DELETE",
}

export const OPTIONS = ["Edit", "Delete"];
