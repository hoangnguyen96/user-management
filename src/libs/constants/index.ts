export * from "./api";
export * from "./dashboard";
export * from "./error-message";
export * from "./mutation-key";
export * from "./query-key";
export * from "./query-key";
export * from "./routes";
export * from "./selection";
export * from "./user";
export * from "./validate";
export * from "./key-token";
