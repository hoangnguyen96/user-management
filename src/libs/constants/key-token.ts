export const TOKEN_KEY = "user-manage";
