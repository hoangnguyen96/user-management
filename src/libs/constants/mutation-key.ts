export const MUTATION_KEY = {
  CREATE_CUSTOMER: ["createCustomer"],
  CREATE_PRODUCT: ["createProduct"],
  LOGOUT: ["logout"],
};
