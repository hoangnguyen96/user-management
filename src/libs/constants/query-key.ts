export const QUERY_KEY = {
  CUSTOMERS_LIST: "customersList",
  CUSTOMERS_LIST_BY_ID: "customersListById",
  PRODUCT_LIST: "productList",
  PRODUCT_LIST_OF_USER: "productListOfUser",
  PRODUCT_LIST_OF_USER_BY_ID: "customersListById",
};
