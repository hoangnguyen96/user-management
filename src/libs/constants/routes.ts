export const ROUTES = {
  LANDING_PAGE: "/",
  LOGIN: "/login",
  SIGNUP: "/sign-up",
  DASHBOARD: "/dashboard",
  PRODUCT: "/product",
  CUSTOMERS: "/customers",
  INCOME: "/income",
  PROMOTE: "/promote",
  HELP: "/help",
};
