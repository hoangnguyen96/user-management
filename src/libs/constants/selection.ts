export const SORT_DATA_CUSTOMERS = [
  {
    value: "default",
    name: "-select-",
  },
  {
    value: "name",
    name: "Name",
  },
  {
    value: "company",
    name: "Company",
  },
  {
    value: "phone",
    name: "Phone",
  },
  {
    value: "email",
    name: "Email",
  },
  {
    value: "country",
    name: "Country",
  },
];

export const SORT_DATA_PRODUCT = [
  {
    value: "default",
    name: "-select-",
  },
  {
    value: "name",
    name: "Name",
  },
  {
    value: "price",
    name: "Price",
  },
  {
    value: "quantity",
    name: "Quantity",
  },
];
