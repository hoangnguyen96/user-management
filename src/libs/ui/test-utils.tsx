import { PropsWithChildren, ReactElement, ReactNode } from "react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { BrowserRouter } from "react-router-dom";
import { RenderOptions, render } from "@testing-library/react";

// ThemeProvider
import { ThemeProvider } from "@mui/material";

// Config
import { themeDefault } from "./themes";

const AllTheProviders = ({ children }: PropsWithChildren) => {
  return <ThemeProvider theme={themeDefault()}>{children}</ThemeProvider>;
};

export const wrapper = ({ children }: { children: ReactNode }) => {
  const queryClient = new QueryClient({
    defaultOptions: {
      queries: {
        retry: false,
      },
    },
  });

  return (
    <ThemeProvider theme={themeDefault()}>
      <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
    </ThemeProvider>
  );
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderWithQueryClient = (
  ui: ReactElement,
  options?: RenderOptions
) => render(ui, { wrapper, ...options });

export const WrapperRouter = ({ children }: { children: ReactNode }) => (
  <BrowserRouter>{children}</BrowserRouter>
);

export { customRender as render };
export { customRenderWithQueryClient as renderWithQueryClient };
