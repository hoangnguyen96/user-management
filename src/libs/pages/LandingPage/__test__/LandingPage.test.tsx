import { render } from "@app/ui/test-utils";
import { fireEvent } from "@testing-library/react";
import { useNavigate } from "react-router-dom";

// Constants
import { ROUTES } from "@app/constants";

// Component
import LandingPageBase from "..";

jest.mock("react-router-dom", () => ({
  useNavigate: jest.fn(),
}));

describe("LandingPageBase component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("Should render snapshot correctly", () => {
    expect(render(<LandingPageBase />)).toMatchSnapshot();
  });

  it("Should handle click login", () => {
    const navigate = jest.fn();
    (useNavigate as jest.Mock).mockReturnValue(navigate);
    const { getByText } = render(<LandingPageBase />);

    fireEvent.click(getByText("Login"));

    expect(getByText("Login")).toBeInTheDocument();
    expect(navigate).toHaveBeenCalledWith(ROUTES.LOGIN);
  });

  it("Should handle click sign up", () => {
    const navigate = jest.fn();
    (useNavigate as jest.Mock).mockReturnValue(navigate);
    const { getByText } = render(<LandingPageBase />);

    fireEvent.click(getByText("SignUp"));

    expect(getByText("SignUp")).toBeInTheDocument();
    expect(navigate).toHaveBeenCalledWith(ROUTES.SIGNUP);
  });
});
