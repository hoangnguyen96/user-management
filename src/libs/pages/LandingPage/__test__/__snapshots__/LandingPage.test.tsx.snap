// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageBase component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiStack-root css-iqlgev-MuiStack-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-vitet8-MuiTypography-root"
        >
          Welcome!
        </h1>
        <p
          class="MuiTypography-root MuiTypography-body1 css-1qh8mz2-MuiTypography-root"
        >
          This user management application
        </p>
        <hr
          class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-3eskvv-MuiDivider-root"
        />
        <div
          class="MuiStack-root css-mgxetk-MuiStack-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1uu91vc-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1uu91vc-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            SignUp
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiStack-root css-iqlgev-MuiStack-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-vitet8-MuiTypography-root"
      >
        Welcome!
      </h1>
      <p
        class="MuiTypography-root MuiTypography-body1 css-1qh8mz2-MuiTypography-root"
      >
        This user management application
      </p>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-flexItem css-3eskvv-MuiDivider-root"
      />
      <div
        class="MuiStack-root css-mgxetk-MuiStack-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1uu91vc-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Login
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1uu91vc-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          SignUp
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
